### PR TITLE
Strip/Freeze code from asaskevich/govalidator and drop dependency

### PR DESCRIPTION
**Pull request by dims**
Please see https://github.com/kubernetes/kube-openapi/issues/512 for context.

Notes:
- Copy the bare minimum code we need from `github.com/asaskevich/govalidator`
- Preserve the license and clearly delineate that this is forked from the above repo
- Copy tests related to the methods we use as well
- Ensure go.mod/sum do not refer to the repo anymore

### CODE REVIEW

**Comment by dims** [comment]
/assign @liggitt 

for [pkg/validation/OWNERS](https://github.com/kubernetes/kube-openapi/blob/master/pkg/validation/OWNERS) approval

**Comment by liggitt** [comment]
+1 to isolate

would it make sense to put under https://github.com/kubernetes/kube-openapi/tree/master/pkg/internal/third_party to keep visibility limited and third party things gathered in one spot?

**Comment by dims** [comment]
@liggitt right! we should do that!

(Done!)

File: pkg/internal/third_party/govalidator/patterns.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package govalidator
+
+import "regexp"
+
+// Basic regular expressions for validating strings
+const (
+	CreditCard string = "^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\\d{3})\\d{11})$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:7]: from https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L8

File: pkg/internal/third_party/govalidator/patterns.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	ISBN10     string = "^(?:[0-9]{9}X|[0-9]{10})$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:8]: from https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L9

File: pkg/internal/third_party/govalidator/patterns.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	ISBN13     string = "^(?:[0-9]{13})$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:9]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L10

File: pkg/internal/third_party/govalidator/patterns.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	Hexcolor   string = "^#?([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:10]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L21

File: pkg/internal/third_party/govalidator/patterns.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	RGBcolor   string = "^rgb\\(\\s*(0|[1-9]\\d?|1\\d\\d?|2[0-4]\\d|25[0-5])\\s*,\\s*(0|[1-9]\\d?|1\\d\\d?|2[0-4]\\d|25[0-5])\\s*,\\s*(0|[1-9]\\d?|1\\d\\d?|2[0-4]\\d|25[0-5])\\s*\\)$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:11]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L22

File: pkg/internal/third_party/govalidator/patterns.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	Base64     string = "^(?:[A-Za-z0-9+\\/]{4})*(?:[A-Za-z0-9+\\/]{2}==|[A-Za-z0-9+\\/]{3}=|[A-Za-z0-9+\\/]{4})$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:12]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L27

File: pkg/internal/third_party/govalidator/patterns.go [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	SSN        string = `^\d{3}[- ]?\d{2}[- ]?\d{4}$`
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:13]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L41

File: pkg/internal/third_party/govalidator/patterns.go [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+	Int        string = "^(?:[-+]?(?:0|[1-9][0-9]*))$"
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/patterns.go:14]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/patterns.go#L18

File: pkg/internal/third_party/govalidator/validator.go [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+	return rxHexcolor.MatchString(str)
+}
+
+// IsRGBcolor check if the string is a valid RGB color in form rgb(RRR, GGG, BBB).
+func IsRGBcolor(str string) bool {
+	return rxRGBcolor.MatchString(str)
+}
+
+// IsCreditCard check if the string is a credit card.
+func IsCreditCard(str string) bool {
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/validator.go:38]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/validator.go#L360

File: pkg/internal/third_party/govalidator/validator.go [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+// IsISBN13 check if the string is an ISBN version 13.
+func IsISBN13(str string) bool {
+	return IsISBN(str, 13)
+}
+
+// IsISBN check if the string is an ISBN (version 10 or 13).
+// If version value is not equal to 10 or 13, it will be check both variants.
+func IsISBN(str string, version int) bool {
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/validator.go:78]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/validator.go#L400

File: pkg/internal/third_party/govalidator/validator.go [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+// IsSSN will validate the given string as a U.S. Social Security Number
+func IsSSN(str string) bool {
+	if str == "" || len(str) != 11 {
+		return false
+	}
+	return rxSSN.MatchString(str)
+}
+
+// ToInt convert the input string or any int type to an integer type 64, or 0 if the input is not an integer.
+func ToInt(value interface{}) (res int64, err error) {
```
Comment (liggitt) [inline pkg/internal/third_party/govalidator/validator.go:144]: https://github.com/asaskevich/govalidator/blob/f61b66f89f4a311bef65f13e575bcf1a2ffadda6/converter.go#L35

**Review by liggitt — COMMENTED** [review]
/lgtm
/approve

left breadcrumbs to the commit and location of anything non-trivial from the original repo

**Pull request merged**